.github/workflows/validate-crd.yaml

```diff
@@ -0,0 +1,52 @@
+name: Check CRD Version Update
+
+on:
+  pull_request:
+    paths:
+      - 'pkg/k8s/apis/cilium.io/client/crds/v1alpha1/*.yaml'
+
+jobs:
+  check-version:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
+
+      - name: Get all changed files
+        id: changed-files
+        uses: jitterbit/get-changed-files@v1
+
+      - name: Check for CRD changes and version update
+        run: |
```

Comment:
This seems more complicated that it has to be. My suggestion would be to follow a simpler approach.
Also, this issue requires changes to the CI that tricky to test.

My suggestion would be to use `git diff` to check the changes, and `act` to test the changes locally.